src/flexmeasures_client/client.py

```diff
@@ -46,10 +47,20 @@ class FlexMeasuresClient:
     session: ClientSession | None = None
 
     def __post_init__(self):
-        if not self.scheme:
-            self.scheme: str = "http" if "localhost" in self.host else "https"
-        if self.ssl is None:
-            self.ssl: bool = False if "localhost" in self.host else True
+        if not re.match(r".+\@.+\..+", self.email):
+            print(self.email)
+            raise ValueError("not an email address format string")
+        if self.api_version not in API_VERSIONS_LIST:
+            raise ValueError(f"version not in versions list: {API_VERSIONS_LIST}")
+        # if ssl then scheme is https.
+        if self.ssl:
+            self.scheme = "https"
+        else:
+            self.scheme = "http"
+        if re.match(r"http\:\/\/|https\:\/\/", self.host):
+            print(self.host)
+            raise ValueError("scheme is inferred from ssl and can not be passed in the host")
```

Comment:
In this case, I think we should try to remove the schema from the host and warn the user that it shouldn't pass the scheme within the host.

Reply:
I think it's better to let it just fail but show why it fails. I don't want dependencies to be created on incorrect usage of the client itself. If users of the FlexMeasures clients want to quality of life improvements like automatically stripping the scheme I think that should happen in their app. We could maybe add this in home assistant for instance.